### DISCUSSION TITLE
Where do I find the latest CIM release in a format that the CIMTool can read?

### DISCUSSION THREAD

**Discussion by nlaws-camus** [cimug-org/CIMTool#55] [Q&A]
Can you please post links for where to find the latest CIM files to use in the CIMTool?

**Comment by ucaiug-admin** [comment]
@nlaws-camus :  First off, welcome to using CIMTool.  Great question.  To make this information universally available to all user's I've updated the project's primary README file.  Please let me know if you have further questions

**Comment by nlaws-camus** [comment]
Just to summarize:
1. Get the latest CIM release from the [CIMUG site](https://cimug.ucaiug.org/CIM%20Model%20Releases/Forms/AllItems.aspx)
2. Install [Enterprise Architect](https://sparxsystems.com/products/ea/) (can get 30 day free trial)
3. Convert the .eap file from step 1 to XMI using Enterprise Architect
4. Import the XMI file into CIMTool

I noticed that the CIM Users Group site says 
>The model is available in two file formats. The .EAP format is for use with Sparx Enterprise Architect and [.XMI](http://en.wikipedia.org/wiki/XMI) format supports other CIM tools (such as [IBM Rational Software Modeler](http://en.wikipedia.org/wiki/IBM_Rational_Software_Modeler) and [CIMTool](http://www.cimtool.org/)).

However, both of the latest files, dated July 4, on the current drafts page are .eap. This appears to be a mistake. Is it possible to get the XMI files posted on the CIM Users Group site? It would be very helpful so that one could get started with the CIMTool without installing commercial software (that only works on Windows). I tried emailing the contact email on the CIM Users Group site (contact@cimug.org) but my email bounced back with "address not found".

**Reply by tviegut** [reply]
Hi Nick,

Catch my comments inlined below.

> Just to summarize:
> 
> 1. Get the latest CIM release from the [CIMUG site](https://cimug.ucaiug.org/CIM%20Model%20Releases/Forms/AllItems.aspx)
> 2. Install [Enterprise Architect](https://sparxsystems.com/products/ea/) (can get 30 day free trial)  
_**You technically do not need to download and install EA if you are going to utilize the ```.eap``` file directly instead of XMI.**_
> 3. Convert the .eap file from step 1 to XMI using Enterprise Architect
> 4. Import the XMI file into CIMTool
> 
> I noticed that the CIM Users Group site says
> 
> > The model is available in two file formats. The .EAP format is for use with Sparx Enterprise Architect and [.XMI](http://en.wikipedia.org/wiki/XMI) format supports other CIM tools (such as [IBM Rational Software Modeler](http://en.wikipedia.org/wiki/IBM_Rational_Software_Modeler) and [CIMTool](http://www.cimtool.org/)).
>_**This above content is very outdated.  We haven't used Rational Rose for years for the CIM (easily 12+ years).  At the time this content was written CIMTool did not support directly using ```.eap``` or ```.eapx``` files.  So I'm sure this is confusing.  This should really be updated on the UCAIug site.**_
> 
> However, both of the latest files, dated July 4, on the current drafts page are .eap. This appears to be a mistake. Is it possible to get the XMI files posted on the CIM Users Group site? It would be very helpful so that one could get started with the CIMTool without installing commercial software (that only works on Windows). I tried emailing the contact email on the CIM Users Group site ([contact@cimug.org](mailto:contact@cimug.org)) but my email bounced back with "address not found".

_**Let me know if you'd like to utilize the ```.xmi``` and I can look into having someone post that out to the folder.  Alternatively, your other option would be to grab a prexisting CIMTool project (see [CIMTool Projects](https://cimug.ucaiug.org/CIM%20Profiles/Forms/AllItems.aspx) where they are in ZIP archives).  These projects are already set up and contain XMI files by default.  These ZIP archives can either be imported directly into CIMTool or can be unzipped and the XMI file in the ```\schema``` folder used directly.  Hope this helps.**_

**Comment by ucaiug-admin** [comment]
@nlaws-camus :  

I thought to provide you with a link to get you up and running with **CIMTool**.  I think you'll find this helpful.

## CIMTool Profiling Tutorial

The following PDF resource is publicly available on the UCAIug website and provides a detailed tutorial on how to use **CIMTool** to create and edit profiles based on the Common Information Model: [CIM & CIMTool Tutorial](https://cimug.ucaiug.org/CIMGroups/CIMInterop/IEEE%20Tutorial/CIM%20Tutorial%20Master%20Slide%20Deck-2013.pdf?Mobile=1&Source=%2FCIMGroups%2FCIMInterop%2F%5Flayouts%2F15%2Fmobile%2Fviewa%2Easpx%3FList%3D19ff6048%2De79f%2D46d7%2D8825%2Ddd1416ad3397%26View%3D022235bd%2D8a45%2D4bc7%2D98fc%2D9145d38d1795%26wdFCCState%3D1)

Sections that may be of particular value include:

- Tutorial Section I - Modeling a Substation pp. 4 - 25 General overview of substation modeling.
- Tutorial Section VI - Information Modeling: pp. 121 - 157
- Tutorial Section XII - CIMTool pp. 270 - 357 Provides an overview of CIMTool features and how to use it for profiling. Note that the references to the download sites are now dated.

Note that some content is dated such as pp. 278-279 where JDK/JRE 1.5 is recommended and the "latest version" and download site.  New versions are now available on this GitHub site.  Also pp. 291-292 on how to export a schema from EA is dated.  The screens in new versions of EA look slightly different.  Other than this most of the content should still be relevant.

**Comment by ucaiug-admin** [comment]
@nlaws-camus :  Just wanted to circle back to see if the material was helpful for your purposes.  Let me know if there was anything else you needed.

**Comment by nlaws-camus** [comment]
@admin-cimug yes very helpful! I have not had time to go through the slide deck (over 400 slides!). My goal is to create a profile for a DERMS platform - starting with some relatively simple examples. I have found several mentions of a Common Distribution Power System Model, 61968-13: is there an XML or XMI implementation of the CDPSM? Is there a list of companies using the CDPSM?

@tviegut Yes having the XMI files on the CIM UG site would be very helpful!  (And it says in this repo's README:
> Note that as of the EA 16.x release, .eap and .eapx files are no longer supported. CIMTool does not currently support the new EA 16.x project file format and instead the XMI schema option should be utilized if using the new release.

**Reply by tviegut** [reply]
> @admin-cimug yes very helpful! I have not had time to go through the slide deck (over 400 slides!). My goal is to create a profile for a DERMS platform - starting with some relatively simple examples. I have found several mentions of a Common Distribution Power System Model, 61968-13: is there an XML or XMI implementation of the CDPSM? Is there a list of companies using the CDPSM?
> 
> @tviegut Yes having the XMI files on the CIM UG site would be very helpful! (And it says in this repo's README:
> 
> > Note that as of the EA 16.x release, .eap and .eapx files are no longer supported. CIMTool does not currently support the new EA 16.x project file format and instead the XMI schema option should be utilized if using the new release.

@nlaws-camus :  Hey Nick.  We can check into someone uploading the XMI.  The reality is you don't really have a need for it based on what you're describing.  The existing `.eap` files that are out there on the UCAIug website should be importable without issue for you in **CIMTool**.  You do not need to convert these to `.eapx` (that's only if you care about multi-language support).  I use `.eap` files myself.  And if you really did want to import an XMI file you can just download one of the pre-existing CIMTool projects.

On a different note.  You stated that you are looking into profiling for a DERMS platform.  I don't know if you are familiar with the **IEC 61968-5** published standard (see [here](https://webstore.iec.ch/publication/60069)) but it sounds like this may be what you're interested in.  The **CIMTool** project that corresponds to it and that has the profiles is publicly available at [UCA Part5 Profiles Ed1.0](https://cimug.ucaiug.org/CIM%20Profiles/UCA-Part5-Profiles-1st-Edition-copyright.zip).  That **CIMTool** project includes the full XMI for the CIM (i.e. in the `\Schema` folder).  Below is a description of what is covered in that international standard.  The published standard itself will explain in more depth the usage of the profiles and the use cases covered.  It is purchasable at a reasonable cost of using it in a product or at a utility:

>**IEC 61968-5:2020** is the description of a set of functions that are needed for enterprise integration of DERMS functions. These exchanges are most likely between a DERMS and a DMS. However, since this is an enterprise integration standard which may leverage IEC 61968-100:2013 for application integration (using web services or JMS) or other loosely-coupled implementations, there are no technical limitations for systems with which a DERMS might exchange information. Also, it should be noted that a DERMS might communicate with individual DER using a variety of standards and protocols such as IEC 61850, IEEE 2030.5, Distribution Network Protocol (DNP), Sunspec Modbus, or perhaps Open Field Message Bus (OpenFMB). One role of the DERMS is to manage this disparity and complexity of communications on the behalf of the system operator. However, the communication to individual DER is out of scope of this standard. Readers should look to those standards to understand communication to individual DER’s smart inverter.<br/>
The scope will be limited to the following use case categories:<br/>
• DER group creation – a mechanism to manage DER in aggregate
• DER group maintenance – a mechanism to add, remove, or modify the members and/or aggregated capabilities of a given group of DER
• DER group deletion – removing an entire group
• DER group status monitoring – a mechanism for quantifying or ascertaining the current capabilities and/or status of a group of DER
• DER group forecast – a mechanism for predicting the capabilities and/or status of a group of DER for a given time period in the future
• DER group dispatch – a mechanism for requesting that specified capabilities of a group of DER be dispatched to the grid
• DER group voltage ramp rate control – a mechanism for requesting that a DER group following a ramp rate curve
• DER group connect/disconnect – a mechanism to request that DER either isolate themselves, or reconnect to the grid as needed. 

To address your other questions.  The [IEC61968-13 Ed2.0](https://webstore.iec.ch/publication/34213) published standard defines the profile to be used for exchanging Dx static network models (i.e. CDPSM).  I do not believe there is currently a **CIMTool** Project archive hosted out on the UCAIug (CIM User's Group site).

For your question:

> "...is there an XML or XMI implementation of the CDPSM?".   

Typically, profiles for either the CPSM or CDPSM are expressed as RDFS schemas.  An example **CIMTool** project based on the CIM16 Transmission CIM can be located at [CPSM_2013_CIM_16-2016-03-16-CIMTool Project.zip](https://cimug.ucaiug.org/CIM%20Profiles/CPSM_2013_CIM_16-2016-03-16-CIMTool%20Project.zip).  This will give you an idea of what this will look like.  If you read the PDF slide deck in depth it will explain how to validate static network model instance data (CIM/XML files) against a profile defined in **CIMTool**.

For your question:

> Is there a list of companies using the CDPSM?

I don't know of a formal list.  I know in Europe that IEC 61968-13 Ed 2.0 has undergone interoperability tests as you will see in the Abstract for the standard below:

>IEC 61968-13:2021 specifies profiles that can be used to exchange Network Models in a Utility or between a Utility and external applications to the utility. This document provides a list of profiles which allow to model balanced and unbalanced distribution networks in order to conduct network analysis (Power flow calculation). Therefore it leverages already existing profiles (IEC 61970-45x based on IEC 61970-301 (CIM base) or profiles based on IEC 61968­11 CIM extension for Distribution). This document reuses some profiles without any change, or eventually extends them or restricts them. Moreover it proposes other profiles to reflect Distribution needs.
Use of CIM in Distribution is not a new topic. This document includes informative parts, as CIM model extensions, which could be integrated in future versions of the IEC CIM Model. These extensions have been used by some utilities for utility internal information exchange use cases and to support information exchanges between different market participants like Transmission System Operators (TSO), Distributed System Operators (DSO), Distributed Network Operators (DNO) and Significant Grid Users (SGU) including generators and industry.<br/>This second edition cancels and replaces the first edition published in 2008. This edition constitutes a technical revision. This edition was pre-tested during 2016 ENTSO-E interoperability tests. The interoperability test report mentions: "Some vendors demonstrated that the transformation between distribution network and CGMES is possible. This is a first step towards the efforts to have closer integration between CGMES and profiles for exchanging distribution data (CDPSM)."

**Comment by tviegut** [comment]
@nlaws-camus :  Hey Nick.  Hopefully you caught my prior response.  Does the additional information for the IEC 61968-5 standard for DERMS relate to the work you were thinking about doing?

**Comment by thisiswhereitype** [comment]
@tviegut To pick up on this - I am trying to analyse the CIM, but found that that project only for CIMtools contains only (I think) 61970.

[Wikipedia ](https://en.wikipedia.org/wiki/Common_Information_Model_(electricity))seems to describe the CIM as including multiple standards some of which are extensions. Should they be included? I am interested in 61968-11for distribution companies.

**Reply by glirellus** [reply]
Are you trying to analyze the base model, or are you trying to analyze the messages that are exchanged? Are you interested in data engineering or are in interested in operations?  
If you are trying to look at the base model, then you will be better served by looking at the UML directly in Enterprise Architect instead of CIMTool because CIMTool is a tool for creating profiles from the base model. You can get the latest Enterprise Architect UML files here: https://cimug.ucaiug.org/CIM%20Model%20Releases/Forms/AllItems.aspx
The UML covers the engineering, operations, and market scope all in the same file.  When you are looking at the IEC standards, the IEC 61970-301 covers the underlying UML engineering network model for both transmission and distribution.  The IEC 61968-11 covers the underlying UML operations model for both transmission and distribution.  

Note that you will only get so far when looking at the base model because it doesn't really tell you how the data is actually used and formed into messages. For that you need to look at message profiles.
If you are interested in the profile for exchanging the engineering network model (for example the data that says that a switch exists), you would want to look at IEC 61970-452 for transmission and IEC 61968-13 for distribution. Note that the UCA site is currently in the middle of a migration so not all of the artifacts are available yet.  One of the older corresponding artifacts for the -452 is at (https://cimug.ucaiug.org/CIM%20Profiles/CPSM_2013_CIM_16-2016-03-16-CIMTool%20Project.zip).  The most recent artifact for -452 has not yet been posted and the artifact for the -13 has also not yet been posted.
If you are interested in the profiles for operations (for example, the data that says that a switching is being opened or closed), you would want to look at all of the rest of the IEC 61968-XX profiles. You can find the newer CIM Tool projects for those profiles at (https://cimug.ucaiug.org/Standards%20Artifacts/Forms/AllItems.aspx?RootFolder=%2FStandards%20Artifacts%2FUCA%20TF%2014&FolderCTID=0x0120001062F2F1DF27704DBB748ABBDC3B3AA2&View=%7BFEBD8EE1%2D6B40%2D42F6%2DB228%2DCCF131291FBE%7D) and the older profiles at (https://cimug.ucaiug.org/CIM%20Profiles/Forms/AllItems.aspx).  Again, the site is in the middle of a migration, so some of the artifacts are moving around or may be missing.

**Reply by thisiswhereitype** [reply]
@glirellus - Thanks for your reply, I am interested to some extent in both. I am considering the CIM among other energy ontology in my research area. I suspect I will want to comment on aspects of all standards.

I don't have access to the EA software. 

> Note that you will only get so far when looking at the base model because it doesn't really tell you how the data is actually used and formed into messages. For that you need to look at message profiles.

I am aware of the difference having read the EPRI primer, I understood I will not be interested in many elements.

Do you know where I should access the underlying RDF, this will contain the profiles and object definitions and I can load it into any software?

**Comment by roachgf85** [comment]
Anyone know how to fix the SSL configuration issue with https://cimug.ucaiug.org/? Cloudflare has been throwing up an SSL handshake failed page for the last week or two since I started looking for the EA models.

**Reply by jarthurgrey** [reply]
@roachgf85 https://cimug.org/ is the new url after some migrations. I hope this helps!

**Comment by roachgf85** [comment]
Thanks Jesse!
